python/spec/dependabot/python/update_checker/requirements_updater_spec.rb

```diff
@@ -129,7 +147,7 @@
           context "when requirement version is too high" do
             let(:requirement_txt_req_string) { ">=2.0.0" }
 
-            its([:requirement]) { is_expected.to eq(:unfixable) }
+            its([:requirement]) { is_expected.to eq(">=1.5.0") }
```

Comment:
I like the previous outcome better here. If the latest resolvable version is 1.5.0, and user has specified >=2.0.0 in their manifest, I think either:

* User setup/manifest is wrong.
* Dependabot run into a bug when figuring out the latest resolvable version.

Either way, it seems best to leave this kind of ill defined case alone?

Also, the new solution is actually widening the requirement, not increasing it, which seems completely against the spirit of this PR?

Reply:
The new solution actually it is not widening it. It is setting the lower bound as the latest version and the upper bound is still following the last version by increasing. As it is mentioned in the [document](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file) tried to follow the typical scenario. 

```
Always increase the minimum version requirement to match the new version. If a range already exists, typically this only increases the lower bound.
```

Reply:
If I'm understanding this test case correctly, the previous requirement is `>= 2.0.0`, correct? If Dependabot sets the new range to be `>= 1.5.0`, it's actually _decreasing_ the lower bound and thus widening the requirement.

This seems like an ill-defined case to me. Why would the existing range be `>= 2.0.0` if the latest version is 1.5.0? That's why it seemed reasonable to not mess with this case and call it "unfixable".

That said, I just came up with a potential case when this could happen in real life: If latest version 2.0.0 is yanked, then that could make 1.5.0 the "new" latest version and make a case like this possible.

Previously Dependabot would avoid messing up with this case, and now it's downgrading the version to 1.5.0. I guess it's not as bad after all!

Reply:
I am strongly of the opinion that dependabot should never lower a `>=` constraint. When such a constraint is used, I believe it means one of two things:
1. We're tracking the latest version: the user should pick the `increase` strategy.
2. We're declaring the range of all compatible versions and we **know** that lower versions are not compatible (e.g. we're using a feature that was introduced in 2.0): user should pick `widen` strategy and dependabot should not touch lower bound.

Reply:
I realize I got the context wrong: we're talkina bout `increase` here. I still think a user intervention would be more appropriate here, but I am not as strongly invested as for the widen case.

Reply:
I agree this should be left untouched and leave it to the user to manually fix, but given the realworld situation where I think this can happen (latest version is removed), I can also see how some people would prefer that Dependabot "downgrades" the lower bound to make manifests functional again. I _think_ this change may also affect the `widen` strategy, by the way?

Reply:
> I can also see how some people would prefer that Dependabot "downgrades" the lower bound to make manifests functional again.

For `increase` I can see that. I'm not 100% sure where I stand myself, but I agree that there is something to say for it.

For `widen` however (assuming that this change does in fact affect it, I've not checked the diff), if a lower bound is present, it will always be user specified, and typically for good reason. Since dependabot does not have any context regarding it, it should be considered outside of dependabot's jurisdiction imo.

Reply:
Yeah, I do think there could be all sort of expectations here. I guess it could also happen that someone using `widen` has changed their code to support 2.0 version of some library and bumped their lower bound requirement to `>=2.0`, then 2.0 is abruptly removed and user may appreciated that Dependabot "rolls back" their requirement change and let them know through a PR they need to revert code changes too to get back on track.

Also, strictly speaking, for the `widen` strategy, Dependabot is also doing exactly what's advertised right? Namely, "widen the range to accommodate for the latest version". Just that in this case, the latest version is not newer than the upper bound, but older than the lower bound.

In any case, I think this is exactly why I advocate for "leave this kind of case alone", because different users will have different expectations, or even the same user may want different things depending on the library.

Reply:
You do make a good point, this is a case where it's difficult to find a universal approach to fit everyone's use case. Your suggestion to leave it alone sounds sensible to me (biased though I am because that is what I've been advocating regardless).

Reply:
When I implemented this, my thoughts were as follows:

In real-world scenarios, this occurs when no range was specified before, when a higher version is yanked and necessitates a downgrade to a previous release, or when the strategy wasn’t set to increase when a lower bound is added. In such cases, Dependabot can adjust the lower bound accordingly.

For the increase strategy, if a lower constraint is defined to ensure compatibility, the lower bound should not be higher than the latest version since the latest version is functional for the app or library. Otherwise, it risks excluding versions that work correctly, potentially breaking the app.

I would appreciate your thoughts on this.

Reply:
I see what you're saying, but I still feel like it's a risky assumption to make for the widen strategy. For example, consider the case where a library detects a breaking bug in their latest release, so they yank it and then release a patched one soon after. If they do it in that order, dependabot might run in the window where the broken one was yanked but the new one not yet released. It would then change the lower bound, while it wouldn't have taken action at all if it had run a few minutes later.

If the library does not release a patched version (which is quite rare as far as I know), then action will indeed be required. But since this is such a rare occurrence, and lowering the constraint might not *always* be desired, I feel like it's safer to leave this as a problem for the user to solve.

That said, I'm just a bystander here, so I won't try to push my view any further than this.

Reply:
I agree with @sanderr here. I think Dependabot should create update PRs that work for everyone, or when that's not the case, at least allow the behavior to be configurable. This seems such an edge case that feels best to not try be too smart and leave it for the user to fix manually.